### PR TITLE
Pin PyYAML for testing

### PR DESCRIPTION
**Pull request by lmazuel**

### CODE REVIEW

**Comment by stefangordon** [comment]
@lmazuel I noticed azure-cli pinned `pyyaml==4.2b4` not sure if that ends up causing a problem?

**Comment by lmazuel** [comment]
@stefangordon Since this PR, I unpinned it here: https://github.com/Azure/azure-sdk-for-python/pull/2841

SDK is using PyYAML for testing only, and once everything is recorded with Py3.x, the version of PyYAML doesn't matter anymore.
CLI is using PyYAML for testing and for help generation. Also, the only version compatible with Py3.7 is PyYAML 4.x. In order to ship CLI for Py3.7, there is no other way than to min the 4.x version.

TL;DR; current status is ok :)

**Comment by stefangordon** [comment]
Got it.  Thank you.

**Comment by stefangordon** [comment]
Oh, I guess they are also releasing 3.13 for 3.12 back compat + 3.7 compat
https://github.com/yaml/pyyaml/issues/126#issuecomment-402194635

Doesn't really change much, but interesting.
